frontend/components/ExplicitInputSwitch.vue

```diff
@@ -0,0 +1,24 @@
+<script setup lang="ts">
+interface Props {
+  onLabel: string
+  offLabel: string
+  value: boolean
+}
+const props = defineProps<Props>()
+
+const emit = defineEmits<(e: 'update:value', value: boolean) => void>()
```

Comment:
optional super nit: define this in an `Emits` interface for readability, makes it easier to parse at a glance what the contract of a component is when types for both props/emits are at the top.